scompose/project/instance.py

```diff
@@ -541,8 +541,9 @@ def create(self, ip_address=None, sudo=False, writable_tmpfs=False):
             # Volumes
             options += self._get_bind_commands()
 
-            # Ports
-            options += self._get_network_commands(ip_address)
+            # Network configuration + Ports
+            if self.network["enable"]:
+                options += self._get_network_commands(ip_address)
 
             # Hostname
             options += ["--hostname", self.name]
```

Comment:
I haven't tried this so I don't know the answer - do we want a hostname if we don't enable network options?

Reply:
Good point. I tried that. It doesn't enable the network options.. I agree that sounds counter intuitive but singularity is funny sometimes, ay?  😂

Reply:
But if you want I can move the `--hostname` under the same `if` statement.

Reply:
I think it's okay to leave - if there is some addition to a resolver file that gives the container a hostname without setting up networking, I suppose someone could want that? And it doesn't hurt to have it. So let's leave as is until there is more evidence to change that.